app/questionnaire/questionnaire_schema.py

```diff
@@ -378,6 +379,22 @@ def _update_answer_dependencies_for_calculated_summary_dependency(
             calculated_summary_block
         )
         for answer_id in calculated_summary_answer_ids:
+            # for any dynamic answer ids based on a list, the calculated summary needs to depend on the remove block for the list
```

Comment:
Minor and not sure how the other devs feel about this, but maybe we could make the doc string a bit more comprehensive to cover the comments added here. Would make it more readable for me but not sure what others think 🤔 Same applies to some of the other method updated elsewhere in this PR.

Reply:
yeah absolutely 👍 I'll happily go back through and make the docstrings a bit stronger

Reply:
Looks good to me! 👍